nengo_spa/algebras/hrr_algebra.py

```diff
@@ -210,3 +342,132 @@ def zero_element(self, d, sidedness=ElementSidedness.TWO_SIDED):
+    the sign is considered negative if either component is negative; and
+    the sign is considered zero if both are zero. Binding two Semantic Pointers
+    with the same sub-sign will yield a positive Semantic Pointer. See the table
+    below for details.
+
+    .. table:: Resulting Semantic Pointer signs from HRR binding two Semantic
+         Pointers. (Only the upper triangle is given as the matrix is
+         symmetric.)
+
+        ================== =========== ========== ========== ========== ======
+        Sign (DC, Nyquist) \+ (+1, ≥0) − (+1, <0) − (-1, ≥0) − (−1, <0) (0, 0)
+        ================== =========== ========== ========== ========== ======
+        \+ (+1, ≥0)        \+ (+1, ≥0) − (+1, <0) − (−1, ≥0) − (−1, <0) (0, 0)
+        − (+1, <0)                     \+ (1, ≥0) − (−1, <0) − (−1, ≥0) (0, 0)
+        − (−1, ≥0)                                \+ (1, ≥0) − (+1, <0) (0, 0)
+        − (−1, <0)                                           \+ (1, ≥0) (0, 0)
+        (0, 0)                                                          (0, 0)
+        ================== =========== ========== ========== ========== ======
+
+    Parameters
+    ----------
+    dc_sign : int
+        Sign of the DC component.
+    nyquist_sign : int
+        Sign of the Nyquist frequency component. Set to zero for odd
+        dimensionalities by definition.
+    """
+
+    __slots__ = ["dc_sign", "nyquist_sign"]
+
+    def __init__(self, dc_sign, nyquist_sign):
+        if dc_sign == 0 and nyquist_sign != 0:
+            raise ValueError(
+                "nyquist_sign must be 0 if dc_sign is 0 to constitute a valid "
+                "sign in the HrrAlgebra."
+            )
+        if dc_sign not in (-1, 0, 1):
+            raise ValueError("dc_sign must be one of -1, 0, 1")
+        if nyquist_sign not in (-1, 0, 1):
+            raise ValueError("nyquist_sign must be one of -1, 0, 1")
+
+        self.dc_sign = dc_sign
+        self.nyquist_sign = nyquist_sign
+
+    def is_positive(self):
+        return self.dc_sign > 0 and self.nyquist_sign >= 0
+
+    def is_negative(self):
+        return self.dc_sign < 0 or self.nyquist_sign < 0
+
+    def is_indefinite(self):
+        return False
+
+    def to_vector(self, d):
+        """Return the vector in the algebra corresponding to the sign.
+
+        ============  =================  ====================================
+        DC component  Nyquist component  Vector
+        ============  =================  ====================================
+         1            >=0                [ 1,  0, 0, ...] (identity)
+         1             <0                [ 0,  1, 0, 0, ...]
+        -1            >=0                [ 0, -1, 0, ...]
+        -1             <0                [-1,  0, 0, 0, ...] (negative identity)
+         0              0                [ 0,  0, 0, ...] (zero)
+        ============  =================  ====================================
```

Comment:
Last question is about this table here and about in the class docstring for `HrrSign`. Why is the DC component assumed to be in `{-1, 0, 1}` while the Nyquist component is floating? I'd think they should both be floating? If it's unitary then they would both be in `{-1, 0, 1}` but that does not need to be the case.

Reply:
Within the sign representation both are in `{-1, 0, 1}` ([see lines 387–395](https://github.com/nengo/nengo-spa/pull/271/files/ee2f7d2a776cd32dffe73af153c7dbcd917303c2#diff-daa201651cf1b7c7487c0daf2ce5b8dc327330530a1cabf826953c623bac3fe1R387-R395)). Using `>=`/`<` notation requires two rows less in the table. I thought that would be easier to read.

Reply:
Oh I see. So am I to read `<0` as `-1` and `>=0` as `{0, 1}`? Confused me a bit because the columns in the table are also "DC component" and "Nyquist component" instead of "DC sign" and "Nyquist sign" (thinking latter would correspond to the parameters with the `{-1, 0, 1}` sign representation, and then the former are the components from which they are derived).

Reply:
I can rename the columns.

Reply:
Sounds good to go to me then. I like all of this much better long-term than what I was doing with the "nondegenerate" stuff in #243. Awesome stuff!